src/native/mac/qgsmacnative.mm

```diff
@@ -0,0 +1,36 @@
+/***************************************************************************
+    qgsmacnative.cpp - abstracted interface to native Mac objective-c
+                             -------------------
+    begin                : January 2014
+    copyright            : (C) 2014 by Larry Shaffer
+    email                : larrys at dakotacarto dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmacnative.h"
+
+#include <Cocoa/Cocoa.h>
+
+QgsMacNative::~QgsMacNative()
+{
+}
+
+const char* QgsMacNative::currentAppLocalizedName()
+{
+  return [[[NSRunningApplication currentApplication] localizedName] UTF8String];
+}
+
+void QgsMacNative::currentAppActivateIgnoringOtherApps()
+{
+  // valid for Mac OS X >= 10.6
+  [[NSRunningApplication currentApplication] activateWithOptions:
+    (NSApplicationActivateAllWindows | NSApplicationActivateIgnoringOtherApps)];
```

Comment:
@dakcarto is this the correct thing to do?

Quoting from [the NSApplicationActivateIgnoringOtherApps docs](https://developer.apple.com/reference/appkit/nsapplicationactivationoptions/nsapplicationactivateignoringotherapps?language=objc)
> You should **rarely pass this flag** because stealing key focus produces a very bad user experience.